src/api/db/models/User.js

```diff
@@ -0,0 +1,83 @@
+import Cognito from '@aws-sdk/client-cognito-identity-provider';
+import { ApolloError, ForbiddenError } from 'apollo-server-errors';
+import { MANAGE_USERS_ROLES } from '../../auth/roles.js';
+import { hasRole } from './utils.js';
+
+/**
+ * Users are managed in AWS Cognito but as the APIs are designed to be similiar
+ * to those of DB objects, the model is defined here
+ */
+export class UserModel {
+
+  /**
+   * List Users part of a given group
+   *
+   * @param {object} input
+   * @param {string} input.filter Filter usernames by string
+   * @param {object} context
+   */
+  static async list(input, context) {
+    const cognito = new Cognito.CognitoIdentityProviderClient();
+
+    try {
+      const users = (await Promise.all(['manager', 'observer', 'member'].map(async (role) => {
+        const list = [];
+        let res = {};
+        do {
+          res = await cognito.send(new Cognito.ListUsersInGroupCommand({
+            UserPoolId: context.config['/APPLICATION/COGNITO/USERPOOLID'],
+            GroupName: `animl/${context.user['curr_project']}/project_${role}`
+          }));
+
+          list.push(...(res.Users || []));
+        } while (res.NextToken);
+
+        return list.map((user) => {
+          user.role = role;
+          return user;
+        });
+      }))).reduce((acc, cur) => {
+        acc.push(...cur);
+        return acc;
+      }, []).map((user) => {
+        const meta = user.Attributes.reduce((acc, cur) => {
+          acc[cur.Name] = cur.Value;
+          return acc;
+        }, {});
+
+        return {
+          role: user.role,
```

Comment:
Just flagging that currently users can have multiple roles within a single project (e.g. they can be both a 'manager' and 'observer'). I'm just doing a quick scan and haven't read the code here closely enough to confirm whether or not you support this.

Reply:
It previously supported this but would return two distinct entries for each "role" a user was part of. I've improved this behavior in a subsequent commit to instead return a roles array and perform a deduplication/aggregation on the project level.